src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py

```diff
@@ -27,12 +27,32 @@
 @pytest.mark.sdk_test
 @pytest.mark.e2etest
 class TestFlowAsFunc:
-    def test_flow_as_a_func(self):
-        f = load_flow(f"{FLOWS_DIR}/print_env_var")
+    @pytest.mark.parametrize(
+        "test_folder",
+        [
+            f"{FLOWS_DIR}/print_env_var",
+            f"{FLOWS_DIR}/print_env_var_async",
+        ],
+    )
+    def test_flow_as_a_func(self, test_folder):
+        f = load_flow(test_folder)
         result = f(key="unknown")
         assert result["output"] is None
         assert "line_number" not in result
 
+    @pytest.mark.asyncio
+    @pytest.mark.parametrize(
+        "async_call_folder",
+        [
+            f"{FLOWS_DIR}/print_env_var",
+            f"{FLOWS_DIR}/print_env_var_async",
+        ],
+    )
+    async def test_flow_as_a_func_asynckw(self, async_call_folder):
+        f = load_flow(async_call_folder, async_call=True)
```

Comment:
What happens if the flow only contains non-async functions?

Reply:
it does no harm.
1. it is the executor that really deal with the non-async function. each node is scheduled as async in theory.
2. by the way, call sync function in async call is total valid scenario.

Reply:
By the way, 
print_env_var is total sync flow
print_env_var_async is total async flow.